eager-workflow-start/main.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"context"
+	"log"
+
+	"github.com/pborman/uuid"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/worker"
+
+	"github.com/temporalio/samples-go/helloworld"
+)
+
+const taskQueueName = "eager_wf_start"
+
+func main() {
+	// 1. Create the shared client.
+	c, err := client.Dial(client.Options{})
+	if err != nil {
+		log.Fatalln("Unable to create client", err)
+	}
+	defer c.Close()
+
+	// 2. Start the worker in a non-blocking manner before the workflow.
+	workerOptions := worker.Options{
+		OnFatalError: func(err error) { log.Fatalln("Worker error", err) },
+	}
+	w := worker.New(c, taskQueueName, workerOptions)
+	w.RegisterWorkflow(helloworld.Workflow)
+	w.RegisterActivity(helloworld.Activity)
+	err = w.Start()
+	if err != nil {
+		log.Fatalln("Unable to start worker", err)
+	}
+	defer w.Stop()
+
+	workflowOptions := client.StartWorkflowOptions{
+		ID:        "eager_wf" + uuid.New(),
+		TaskQueue: taskQueueName,
+
+		// 3. Set this flag to true to enable EWS.
+		EnableEagerStart: true,
```

Comment:
Hrmm, wonder if it's a bit much to have an entire sample just to set true on a single workflow option (I wouldn't expect separate samples for every other start option). But I don't mind I guess if it makes sense generally.

Reply:
I think it is worth making a sample for since it also shows you have to reuse the client